pkg/webhook/genericsimpleevent.go

```diff
@@ -55,57 +50,46 @@ func (g *genericWebhook) Handle(c *gin.Context) {
 		return
 	}
 
-	gwData := &genericWebhookData{}
-
-	err = c.BindJSON(gwData)
+	payload, err := ioutil.ReadAll(c.Request.Body)
 	if err != nil {
-		log.Printf("Failed to convert POST data into JSON: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed POST data"})
+		log.Printf("Failed to read body: %s", err)
+		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed body"})
 		return
 	}
+	defer c.Request.Body.Close()
 
-	body, err := ioutil.ReadAll(c.Request.Body)
-
+	// try to unmarshal Revision data, if they do exist
+	revision := &brigade.Revision{}
+	err = json.Unmarshal(payload, &revision)
 	if err != nil {
-		log.Printf("Failed to read body: %s", err)
-		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed body"})
+		log.Printf("Failed to convert POST data into JSON: %s", err)
+		c.JSON(http.StatusBadRequest, gin.H{"status": "Malformed POST data - Invalid JSON"})
 		return
 	}
-	defer c.Request.Body.Close()
 
-	go g.notifyGenericWebhookEvent(proj, body, gwData)
-	c.JSON(200, gin.H{"status": "Success"})
+	go g.notifyGenericWebhookSimpleEvent(proj, payload, revision)
+	c.JSON(200, gin.H{"status": "Success. Build created"})
 }
 
-func (g *genericWebhook) notifyGenericWebhookEvent(proj *brigade.Project, payload []byte, gwData *genericWebhookData) {
-	if err := g.genericWebhookEvent(proj, payload, gwData); err != nil {
-		log.Printf("failed genericWebhook event: %s", err)
+func (g *genericWebhookSimpleEvent) notifyGenericWebhookSimpleEvent(proj *brigade.Project, payload []byte, revision *brigade.Revision) {
+	if err := g.genericWebhookSimpleEvent(proj, payload, revision); err != nil {
+		log.Printf("failed genericWebhook SimpleEvent: %s", err)
 	}
 }
 
-func (g *genericWebhook) genericWebhookEvent(proj *brigade.Project, payload []byte, gwData *genericWebhookData) error {
-	revision := &brigade.Revision{}
-	revision.Commit = gwData.Commit
-	revision.Ref = gwData.Ref
-
-	// get brigade.js
-	script, err := GetFileContents(proj, revision.Ref, "brigade.js")
-	if err != nil {
-		log.Printf("Error getting file: %s", err)
-		return fmt.Errorf("Error getting file: %s", err)
-	}
-
-	// create a Build for the specified Revision
+func (g *genericWebhookSimpleEvent) genericWebhookSimpleEvent(proj *brigade.Project, payload []byte, revision *brigade.Revision) error {
 	b := &brigade.Build{
 		ProjectID: proj.ID,
-		Type:      "webhook",
+		Type:      "simpleevent",
 		Provider:  "GenericWebhook",
 		Payload:   payload,
 		Revision:  revision,
-		Script:    script,
 	}
-	if proj.DefaultScript != "" {
-		b.Script = []byte(proj.DefaultScript)
+
+	// set a default Revision if user has not provided any information about commit or ref
+	if b.Revision == nil || (b.Revision.Commit == "" && b.Revision.Ref == "") {
+		b.Revision = &brigade.Revision{Ref: "master"}
```

Comment:
Is this actually needed?

Reply:
I think so. If no Commit or Ref was specified, I had an issue on a GitHub repo I used (and this was the reason I encountered #810, hehe). Let me give it one more shot **without** including set to 'master' to make sure

Reply:
Indeed, just tested with these lines commented out, running a build against `deis/empty-testbed`. Sidecar fails to initialize, with the following error
```
+ :
+ :
+ : /vcs
+ refspec=
+ cut -f2
+ git ls-remote --exit-code https://github.com/deis/empty-testbed.git
+ full_ref=
+ git init -q /vcs
+ cd /vcs
+ retry git fetch -q --force --update-head-ok https://github.com/deis/empty-testbed.git
+ local 'n=1'
+ local 'max=5'
+ local 'delay=5'
+ true
+ git fetch -q --force --update-head-ok https://github.com/deis/empty-testbed.git
+ break
+ retry git checkout -q --force
+ local 'n=1'
+ local 'max=5'
+ local 'delay=5'
+ true
+ git checkout -q --force
fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
Command failed. Attempt 1/5. Waiting for 5 seconds before retrying.
+ test 1 -lt 5
+ echo 'Command failed. Attempt 1/5. Waiting for 5 seconds before retrying.'

...
sidecar retries but ultimately fails :(
```

Setting the `master` to the Ref did the trick. Can you also check, maybe fault is something in my configuration?

Reply:
No, you are right - I think we should let it like this.